Extensions/Lighting/tests/lightruntimeobject.spec.js

```diff
@@ -0,0 +1,355 @@
+      new Float32Array([100, 300, 300, 300, 300, 100, 100, 100])
+    );
+    expect(gdjs.LightRuntimeObjectPixiRenderer._defaultIndexBuffer).to.eql(
+      new Float32Array([0, 1, 2, 0, 2, 3])
+    );
+  });
+});
+
+describe('Light with obstacles around it', function () {
+  const runtimeGame = new gdjs.RuntimeGame({
+    variables: [],
+    resources: {
+      resources: [],
+    },
+    properties: { windowWidth: 800, windowHeight: 600 },
+  });
+  const runtimeScene = new gdjs.RuntimeScene(runtimeGame);
+  runtimeScene.loadFromScene({
+    layers: [{ name: '', visibility: true, effects: [] }],
+    variables: [],
+    behaviorsSharedData: [],
+    objects: [],
+    instances: [],
+  });
+  runtimeScene._timeManager.getElapsedTime = function () {
+    return (1 / 60) * 1000;
+  };
+  const light = addLightObject(runtimeScene, 100);
+  const obstacle = addLightObstacle(runtimeScene, 50, 50);
+
+  it('Vertex and index buffers when light obstacle is present.', function () {
+    light.setPosition(200, 200);
+    obstacle.setPosition(250, 250);
+
+    runtimeScene.renderAndStep();
+    light.update();
+
+    const vertexBuffer = light._renderer._vertexBuffer;
+    const indexBuffer = light._renderer._indexBuffer;
+    const vertexData = [
+      200,
+      200,
+      100,
+      100.0199966430664,
+      100,
+      100,
+      100.0199966430664,
+      100,
+      299.9800109863281,
+      100,
+      300,
+      100,
+      300,
+      100.0199966430664,
+      300,
+      249.9875030517578,
+      300,
+      250,
+      299.9750061035156,
+      250,
+      250.00999450683594,
+      250,
+      250,
+      250,
+      250,
+      250.00999450683594,
+      250,
+      299.9750061035156,
+      250,
+      300,
+      249.9875030517578,
+      300,
+      100.0199966430664,
+      300,
+      100,
+      300,
+      100,
+      299.9800109863281,
+    ];
+    const indexData = [
+      0,
+      1,
+      2,
+      0,
+      2,
+      3,
+      0,
+      3,
+      4,
+      0,
+      4,
+      5,
+      0,
+      5,
+      6,
+      0,
+      6,
+      7,
+      0,
+      7,
+      8,
+      0,
+      8,
+      9,
+      0,
+      9,
+      10,
+      0,
+      10,
+      11,
+      0,
+      11,
+      12,
+      0,
+      12,
+      13,
+      0,
+      13,
+      14,
+      0,
+      14,
+      15,
+      0,
+      15,
+      16,
+      0,
+      16,
+      17,
+      0,
+      17,
+      18,
+      0,
+      18,
+      1,
+    ];
+
+    vertexData.forEach((val, index) => {
+      expect(vertexBuffer[index]).to.be(val);
+    });
+    indexData.forEach((val, index) => {
+      expect(indexBuffer[index]).to.be(val);
+    });
+  });
+
+  it('Vertex and index buffers after obstacle is moved.', function () {
+    obstacle.setPosition(150, 250);
+    runtimeScene.renderAndStep();
+    light.update();
+
+    const vertexBuffer = light._renderer._vertexBuffer;
+    const indexBuffer = light._renderer._indexBuffer;
+    const vertexData = [
```

Comment:
You can add `// prettier-ignore` on the lines just before where you declare arrays to avoid having them formatted to a huge number of lines :) 
See https://prettier.io/docs/en/ignore.html